WIP: Helm (with all other PRs merged)

nwmac opened the pull request.

hamzahamidi commented:
For a  more optimal solution, I was thinking If we could have a JSON as an input instead of using ng-content for code-block-component.ts. We could use this instead 
 ```
pre{
    margin: 0;
    white-space: pre;
  }
```
Here is a screenshot with a fake JSON
![screenshot from 2018-01-19 17-07-37](https://user-images.githubusercontent.com/22576950/35159683-4cc6c276-fd3b-11e7-9729-06cd2e19bea5.png)

KlapTrap commented:
Hi @hamzahamidi. I've added a comment to the original issue to clarify the issue, would you mind taking a look? Thanks a lot.

KlapTrap commented:
Ah, my mistake! Thanks for fixing it. @hamzahamidi

Commit:
Update core style import path

Commit:
Import path changes

Commit:
Update theme import path

Commit:
Even more import updates

Commit:
Imports? Yep!

Commit:
Fix scss font imports

Commit:
Fix more angular json paths

Commit:
More import path fixes

Commit:
Imports

Commit:
Mooore import fixes

Commit:
Last round of import fixes?

Commit:
Merged extensions-phases into monorepo

Commit:
Remove custom symlinks

Commit:
Fix imports after merge

Commit:
Remove symlinks

Commit:
Move the cf cell health list datasource

Commit:
Fix package.json issues

Commit:
Remove unused project from angular.json

Commit:
Linting fix

Commit:
Fix test imports

Commit:
Fix FE unit test imports

Commit:
Fix test

Commit:
Fixed core tests

Commit:
Fix stratos build

nwmac committed:
WIP: Kube dashboard proxy

nwmac committed:
Merge remote-tracking branch 'origin/v2-master' into kube-dashboard

richard-cox committed:
Merge remote-tracking branch 'origin/v2-master' into v3-docs-update

Commit:
Stop paginationResetPagination from added invalid fields

Commit:
Merge branch 'v2-master' into monorepo

* v2-master: (639 commits)
  Stop paginationResetPagination from added invalid fields
  Update status_updates.md
  Updates
  Patch the image version in the chart, so reuse-values can work
  Fix upgrade when using --reuse-values
  Avoid need to change service name
  Add test for upgrading with --reuse-values
  Fix for --reuse-values
  Fix helm repo link when geneating an RC (#3369)
  Improve handling of maxed state and observables (#3370)
  Update cf-api-v3.md
  v2/v3 api docs updates
  Fix create-services script used to set CF up for E2E tests
  Only prevent copy & paste in confirm dialog for production
  Fix for bug causing intermittment E2E Test failures on org tests (#3366)
  Allow e2e createService 3 attempts at creating service
  Weekly updaye 18 January (#3365)
  Change log for 2.3.0 (#3364)
  Ensure the loading indicator and the refresh button animated at the correct time. (#3363)
  Fix fetching of stats in app wall for some cases when list is maxed (#3360)
  ...

Commit:
Changes

Commit:
Move new file into correct packages

richard-cox committed:
WIP

nwmac committed:
Fixes to get customizations working again

richard-cox committed:
Merge remote-tracking branch 'origin/v2-master' into monorepo

richard-cox committed:
Merge fixes

richard-cox committed:
Merge remote-tracking branch 'origin/monorepo' into monorepo

richard-cox committed:
Fix favourites not showing until search value changes
- strange how this does not happen in v2-master

richard-cox committed:
Fix linting

richard-cox committed:
Update package-lock

richard-cox committed:
Merge remote-tracking branch 'origin/v2-master' into monorepo

richard-cox committed:
Merge fixes

richard-cox committed:
Jenkins bump

richard-cox committed:
Fix import

richard-cox committed:
Ensure unit tests run

richard-cox committed:
Test fixes

richard-cox committed:
Update karma.conf.js, specifically reporting and order

richard-cox committed:
Add v3 requirements to for specific endpoints (app, space)

richard-cox committed:
Merge remote-tracking branch 'origin/v2-master' into monorepo

richard-cox committed:
Merge fixes

richard-cox committed:
Common ts files

richard-cox committed:
Fix linting

richard-cox committed:
common karma.conf & test fixes

Commit:
Tidy user service up actions

richard-cox committed:
Merge remote-tracking branch 'origin/v2-master' into monorepo

richard-cox committed:
Merge fixes

richard-cox committed:
Add missing file

Commit:
Add base step component route to service instance creation

Commit:
Remove unused component

Commit:
Ensure we have the correct cancel URL when editing a space service insta

richard-cox committed:
Tweaks

richard-cox committed:
Add separate frontend unit test travis jobs

Commit:
Remove console.log

richard-cox committed:
fix postinstall hook and codecov file path

richard-cox committed:
Fix e2e imports

richard-cox committed:
fix codecov command

richard-cox committed:
ensure tests actual emit code coverage

richard-cox committed:
Remove core-e2e

richard-cox committed:
Remove old e2e cf setup script

Commit:
Added tile selector

richard-cox committed:
Use a single test target instead of multiple
- runs package tests in serial and will stop at package with a failing test
- for discussion

Commit:
Add help message and tidy up tile selector

richard-cox committed:
snyk bump

Commit:
Allow a steppers step to hiden next button

Commit:
Make app name unique directive generic

richard-cox committed:
Angular 7 Update - deps #1

richard-cox committed:
Angular 7 Update - deps #2

Commit:
Fixed create user service

richard-cox committed:
ofType

richard-cox committed:
Package fun

richard-cox committed:
Lint update

Commit:
WIP Multi entity/action lists

Commit:
Put service instance and user provided in same list

Commit:
Fix delete for multi action lists

Commit:
Add updating for user provided services

richard-cox committed:
lint fixes - core

richard-cox committed:
lint fixes - store

richard-cox committed:
Update tslint version to fix exception thrown when linting

Commit:
Fix edit for local lists

richard-cox committed:
Lint fixes

richard-cox committed:
Fix columns issue

richard-cox committed:
Fix exception on endpoints page

richard-cox committed:
Fix styling
- white line on right of side nav
- reduce size and weight of new mahoosive mat-card-title

richard-cox committed:
Template fixes

richard-cox committed:
Unit test fix

richard-cox committed:
Fix store tests

Commit:
Added entity type select to multi action lists

Commit:
Fix provided service delete

richard-cox committed:
WIP

Commit:
Added default select text

richard-cox committed:
Revert updates

richard-cox committed:
Merge remote-tracking branch 'origin/v2-master' into angular7

richard-cox committed:
Merge fixes

richard-cox committed:
Further merge fixes

richard-cox committed:
Fix travis extra unit test, update package-lock, fix unit test

richard-cox committed:
e2e test fix (card action menus), fixed enum component

richard-cox committed:
Fix code coverage

richard-cox committed:
e2e tweak

nwmac committed:
Merge remote-tracking branch 'origin/monorepo' into v2-master

Commit:
Pass schema type into card component

Commit:
WIP Multiple careds per list

richard-cox committed:
Fix exception thrown on home page
- Only happens if there's a favourited entity in a non favourited endpoint and endpoint guid contains a hypen
- Exception was..
  ```
  core.js:1671 ERROR TypeError: Cannot read property 'cnsi_type' of undefined
    at MapSubscriber.project (user-favorite-manager.ts:129)
  ```

richard-cox committed:
Merge branch 'fix-fav-hypen-cf-guid' into endpoint-list-cards

richard-cox committed:
WIP

richard-cox committed:
Fix ctor

richard-cox committed:
Merge branch 'fix-fav-hypen-cf-guid' into endpoint-list-cards

richard-cox committed:
WIP

Commit:
Shwo more information on user provided service instance card

Commit:
Fix update user provided action

richard-cox committed:
Relations fixes
- fixed include relations list for GetAllUserProvidedServices
- fixed schema --> entity typo in api.effect
- use correct space schema in UserProvidedServiceInstanceSchema
- ensure validateEntityRelations uses __forcedPageSchemaKey__
- if present validate the entity type's page number in pagination-reducer.helper

richard-cox committed:
Actions

richard-cox committed:
Add endpoint type icons & custom endpoint details table column/card section

richard-cox committed:
Fix unit tests

Commit:
Fix user service update

richard-cox committed:
WIP

richard-cox committed:
WIP

Commit:
Multi action can now render multiple cards

Commit:
Fix none multi list local list

Commit:
Fix none multi component, multi action lists

Commit:
Fix multi list tables

richard-cox committed:
Fix tests

Commit:
Fix paginator subscription leak

richard-cox committed:
Tweak cf details

Commit:
Remove the need for pageSubscription in list-data-source

richard-cox committed:
Fix linting

Commit:
Improve isPageReady logic

richard-cox committed:
Fix existing getIconButton bug
- There may be other places where this magically worked and now correctly won't

Commit:
Provide maxed entity per local page

richard-cox committed:
Merge remote-tracking branch 'upstream/endpoint-list-cards' into endpoint-cards

richard-cox committed:
Missing images

richard-cox committed:
Merge remote-tracking branch 'upstream/endpoint-list-cards' into endpoint-cards

richard-cox committed:
Add k8s endpoint card image

Commit:
Make maxList work with multi action lists

richard-cox committed:
Disable endpoint card routerLink using new directive

richard-cox committed:
Fix linting, don't show image if no image

richard-cox committed:
Another lint fix

nwmac committed:
Merge pull request #3420 from cloudfoundry-incubator/fix-fav-hypen-cf-guid

Fix exception thrown on home page

nwmac committed:
Merge pull request #3414 from cloudfoundry-incubator/v3-docs-update

Add v3 requirements for specific endpoints (app, space)

Commit:
Revert app wall to single action list

Commit:
Seperate service instance cards

Commit:
CC fixes

Commit:
cc fixes

Commit:
Revert app wall maxlist

richard-cox committed:
v2-master merge

richard-cox committed:
Fix for older cf's with no tags

richard-cox committed:
Missing file following merge

richard-cox committed:
Merge remote-tracking branch 'origin/v2-master' into user-services

# Conflicts:
#	src/frontend/packages/core/sass/_all-theme.scss
#	src/frontend/packages/core/src/core/cf-api-svc.types.ts
#	src/frontend/packages/core/src/features/applications/app-name-unique.directive/app-name-unique.directive.spec.ts
#	src/frontend/packages/core/src/features/applications/app-name-unique.directive/app-name-unique.directive.ts
#	src/frontend/packages/core/src/features/applications/application-wall/application-wall.component.ts
#	src/frontend/packages/core/src/features/applications/create-application/create-application.module.ts
#	src/frontend/packages/core/src/features/applications/edit-application/edit-application.component.ts
#	src/frontend/packages/core/src/features/cloud-foundry/cf.helpers.ts
#	src/frontend/packages/core/src/features/services/services.routing.ts
#	src/frontend/packages/core/src/shared/components/add-service-instance/add-service-instance/add-service-instance.component.html
#	src/frontend/packages/core/src/shared/components/add-service-instance/add-service-instance/add-service-instance.component.ts
#	src/frontend/packages/core/src/shared/components/add-service-instance/csi-mode.service.ts
#	src/frontend/packages/core/src/shared/components/add-service-instance/specify-details-step/specify-details-step.component.ts
#	src/frontend/packages/core/src/shared/components/list/data-sources-controllers/list-data-source-config.ts
#	src/frontend/packages/core/src/shared/components/list/data-sources-controllers/list-data-source-types.ts
#	src/frontend/packages/core/src/shared/components/list/data-sources-controllers/list-data-source.ts
#	src/frontend/packages/core/src/shared/components/list/data-sources-controllers/list-pagination-controller.ts
#	src/frontend/packages/core/src/shared/components/list/data-sources-controllers/local-filtering-sorting.ts
#	src/frontend/packages/core/src/shared/components/list/data-sources-controllers/local-list-controller.ts
#	src/frontend/packages/core/src/shared/components/list/list-cards/card/card.component.html
#	src/frontend/packages/core/src/shared/components/list/list-cards/card/card.component.ts
#	src/frontend/packages/core/src/shared/components/list/list-cards/cards.component.html
#	src/frontend/packages/core/src/shared/components/list/list-cards/cards.component.ts
#	src/frontend/packages/core/src/shared/components/list/list-table/table-cell/table-cell.component.ts
#	src/frontend/packages/core/src/shared/components/list/list-table/table.component.html
#	src/frontend/packages/core/src/shared/components/list/list-types/app/cf-apps-data-source.ts
#	src/frontend/packages/core/src/shared/components/list/list-types/base-cf/base-cf-list-config.ts
#	src/frontend/packages/core/src/shared/components/list/list-types/cf-routes/cf-routes-list-config-base.ts
#	src/frontend/packages/core/src/shared/components/list/list-types/cf-routes/cf-routes-list-config.service.ts
#	src/frontend/packages/core/src/shared/components/list/list-types/cf-services/cf-service-instances-list-config.base.ts
#	src/frontend/packages/core/src/shared/components/list/list-types/cf-services/cf-services-data-source.ts
#	src/frontend/packages/core/src/shared/components/list/list-types/cf-spaces/cf-spaces-list-config.service.ts
#	src/frontend/packages/core/src/shared/components/list/list-types/services-wall/service-instance-card/service-instance-card.component.html
#	src/frontend/packages/core/src/shared/components/list/list-types/services-wall/service-instance-card/service-instance-card.component.ts
#	src/frontend/packages/core/src/shared/components/list/list-types/services-wall/service-instances-wall-data-source.ts
#	src/frontend/packages/core/src/shared/components/list/list-types/services-wall/service-instances-wall-list-config.service.ts
#	src/frontend/packages/core/src/shared/components/list/list.component.html
#	src/frontend/packages/core/src/shared/components/list/list.component.ts
#	src/frontend/packages/core/src/shared/components/list/list.component.types.ts
#	src/frontend/packages/core/src/shared/components/list/list.types.ts
#	src/frontend/packages/core/src/shared/components/schema-form/schema-form.component.ts
#	src/frontend/packages/core/src/shared/components/stepper/step/step.component.ts
#	src/frontend/packages/core/src/shared/components/stepper/steppers/steppers.component.html
#	src/frontend/packages/core/src/shared/components/stepper/steppers/steppers.component.ts
#	src/frontend/packages/core/src/shared/data-services/service-action-helper.service.ts
#	src/frontend/packages/core/src/shared/monitors/pagination-monitor.ts
#	src/frontend/packages/core/src/shared/shared.module.ts
#	src/frontend/packages/store/src/actions/application.actions.ts
#	src/frontend/packages/store/src/actions/list.actions.ts
#	src/frontend/packages/store/src/actions/pagination.actions.ts
#	src/frontend/packages/store/src/actions/router.actions.ts
#	src/frontend/packages/store/src/effects/api.effects.ts
#	src/frontend/packages/store/src/effects/request.effects.ts
#	src/frontend/packages/store/src/helpers/entity-factory.ts
#	src/frontend/packages/store/src/helpers/entity-relations/entity-relations.tree.ts
#	src/frontend/packages/store/src/helpers/entity-relations/entity-relations.ts
#	src/frontend/packages/store/src/helpers/paginated-request-helpers.ts
#	src/frontend/packages/store/src/helpers/schema-tree-traverse.ts
#	src/frontend/packages/store/src/reducers/api-request-reducer/succeed-request.ts
#	src/frontend/packages/store/src/reducers/api-request-reducer/types.ts
#	src/frontend/packages/store/src/reducers/api-request-reducers.generator.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-clear-pagination-type.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-create-pagination.spec.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-max-reached.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-reset-pagination.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-set-client-page-size.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-set-client-page.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-set-page.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-start.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-success.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-update.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer.failure.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer.helper.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination.reducer.ts
#	src/frontend/packages/store/src/types/entity.types.ts
#	src/frontend/packages/store/src/types/pagination.types.ts
#	src/frontend/packages/store/src/types/request.types.ts

richard-cox committed:
Remove junk after merge

nwmac committed:
Merge remote-tracking branch 'origin/v2-master' into weekly-update

nwmac committed:
Updaes

nwmac committed:
Merge pull request #3424 from cloudfoundry-incubator/weekly-update

Weekly update

richard-cox committed:
Fix new lint error

Commit:
Fix list view switching

richard-cox committed:
Update cf-api-v3.md

Fixed rendering in github, typo and minor tweak

Commit:
Pass entity key to the cell component

Commit:
Fix table loading issue

Commit:
Fix linting

nwmac committed:
Add support for GKE

Commit:
Update e2e tests wirth new create service instance flow.

richard-cox committed:
Fix service broker action
- Ensure that the guid property is properly set
- This means error handling happens correcty

nwmac committed:
Merge pull request #3427 from cloudfoundry-incubator/fix-service-broker-action

Fix service broker action

nwmac committed:
Add support for endpoint sub type (and metadata) (#3423)

* Add support for endpoint sub type (and metadata)

* Fix issue with upgrade

nwmac committed:
Fix unit test for new auth form component

Commit:
Fix unit tests

Commit:
Lint fix

richard-cox committed:
Merge remote-tracking branch 'origin/v2-master' into v3-docs-update

nwmac committed:
Remove index.html file

richard-cox committed:
Priorities & tweaks

nwmac committed:
Fix missing .entity

Commit:
More unit test fixes

richard-cox committed:
Merge branch 'angular7' into user-services

# Conflicts:
#	src/frontend/packages/core/src/features/applications/app-name-unique.directive/app-name-unique.directive.ts
#	src/frontend/packages/core/src/shared/components/add-service-instance/specify-details-step/specify-details-step.component.ts
#	src/frontend/packages/core/src/shared/components/list/data-sources-controllers/list-data-source-types.ts
#	src/frontend/packages/core/src/shared/components/list/list-cards/card/card.component.ts
#	src/frontend/packages/core/src/shared/components/list/list-cards/cards.component.ts
#	src/frontend/packages/core/src/shared/components/list/list-table/app-table-cell-default/app-table-cell-default.component.ts
#	src/frontend/packages/core/src/shared/components/list/list-table/table-cell/table-cell.component.ts
#	src/frontend/packages/core/src/shared/components/list/list-types/services-wall/service-instances-wall-list-config.service.ts
#	src/frontend/packages/core/src/shared/components/list/list.component.ts
#	src/frontend/packages/core/src/shared/components/schema-form/schema-form.component.ts
#	src/frontend/packages/core/src/shared/components/stepper/step/step.component.ts
#	src/frontend/packages/core/src/shared/components/stepper/steppers/steppers.component.ts
#	src/frontend/packages/core/src/shared/data-services/service-action-helper.service.ts
#	src/frontend/packages/store/src/actions/router.actions.ts
#	src/frontend/packages/store/src/effects/api.effects.ts
#	src/frontend/packages/store/src/helpers/entity-relations/entity-relations.tree.ts
#	src/frontend/packages/store/src/helpers/entity-relations/entity-relations.ts
#	src/frontend/packages/store/src/reducers/api-request-reducers.generator.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-create-pagination.spec.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-update.ts
#	src/frontend/packages/store/src/reducers/pagination-reducer/pagination.reducer.ts

richard-cox committed:
Link fixes

richard-cox committed:
Merge branch 'user-services-temp' into user-services

richard-cox committed:
Temp remove [appApplicationNameUnique]="appNameChecking"

richard-cox committed:
Remove depentent attribute

nwmac committed:
Add note

richard-cox committed:
Register Endpoint Tile - basic icons

nwmac committed:
Merge pull request #3418 from cloudfoundry-incubator/angular7

Angular 7 Upgrade

nwmac committed:
Merge remote-tracking branch 'origin/v2-master' into endpoint-list-cards

nwmac committed:
Fix lint issues

richard-cox committed:
More info

richard-cox committed:
Merge remote-tracking branch 'origin/v2-master' into endpoint-register-tiles

richard-cox committed:
fix imports

richard-cox committed:
Merge remote-tracking branch 'origin/endpoint-list-cards' into endpoint-register-tiles

richard-cox committed:
Fix issue when combining endpoint list cards & user provided service

nwmac committed:
Fix merge issue

richard-cox committed:
Add tile selector img & tidyup

richard-cox committed:
Merge remote-tracking branch 'upstream/v2-master' into merge-upstream

nwmac committed:
Fix merge issue

richard-cox committed:
Fix linting of custom, and fix lint errors in custom

richard-cox committed:
Fix log in logo

richard-cox committed:
Fix issues from review
- Ensure extensions/extension-types.ts remains free from other imports
- Now check endpoint detail type is correct at runtime
- Cf Endpoint Type is now registered in cf package
- Ensure cf package tests & lint execute correctly
- Ensure endpoint details components are destroyed correctly

richard-cox committed:
Fix unit test

richard-cox committed:
docs update

nwmac committed:
Merge pull request #3422 from cloudfoundry-incubator/endpoint-list-cards

Add card view to endpoint list

nwmac committed:
Helm Chart support (Monocular integration)

nwmac committed:
Merge branch 'v2-master' of github.com:SUSE/stratos into v2-master

nwmac committed:
Merge pull request #3429 from cloudfoundry-incubator/v3-docs-update

CF V3 API docs update

richard-cox committed:
Merge remote-tracking branch 'upstream/v2-master' into merge-upstream

richard-cox committed:
Post merge fixes

richard-cox committed:
Fix endpoint table --> card error

richard-cox committed:
Merge remote-tracking branch 'origin/merge-upstream' into endpoint-cards

# Conflicts:
#	src/frontend/packages/core/src/shared/components/list/list-types/endpoint/endpoint-card/endpoint-card.component.html

nwmac committed:
Add support for slide-ion side help and write in to endpoint connect dialog

nwmac committed:
Fix code climate issues

nwmac committed:
Fix code climate isssues

nwmac committed:
Fix code issue

nwmac committed:
Fix tidy up bug

richard-cox committed:
Merge remote-tracking branch 'upstream/endpoint-register-tiles' into endpoint-subtypes

nwmac committed:
Merge remote-tracking branch 'origin/kube-dashboard' into helm

richard-cox committed:
Add frontend endpoint sub types, wire up to k8s types
- Note - Missing k8s images pending review

nwmac committed:
Update icon font

nwmac committed:
Merge remote-tracking branch 'origin/merge-upstream' into v2-master

nwmac committed:
Merge remote-tracking branch 'origin/endpoint-cards' into v2-master

nwmac committed:
All updates

nwmac committed:
Fix lint issue

nwmac committed:
Update sub type for GKE

nwmac committed:
Merge all latest PRs

nwmac committed:
Merge remote-tracking branch 'upstream/icons-update' into helm-latest

nwmac committed:
Update icons

nwmac committed:
Merge endpoint connect help

richard-cox committed:
Fixed lint error, tweaked naming and access

richard-cox committed:
Allow endpoint types to reference common core endpoint auth types
- stops references of package code in common code
- prep for moving metrics into own package (just need to register metricType obj via decorators)

richard-cox committed:
Add images for k8s

nwmac committed:
Latest and greatest

nwmac committed:
Kube dashboard

nwmac committed:
Fixes

nwmac committed:
Updates

nwmac committed:
Fix backend build script

Commit:
Fix helm module

Commit:
Add favorite to kube endpoint

Commit:
Fix endpoint card favorite star

Commit:
Remove unneeded favorite config

nwmac committed:
Add API to determine if kube dashboard is installed

nwmac committed:
Merge branch 'helm-latest' of github.com:SUSE/stratos into helm-latest

Commit:
Add base kubernetes summary view

Commit:
WIP Kube summary charts

Commit:
Added simple usage chart and pod usage

Commit:
Add status graphs

Commit:
Little bit of tidy up on kube summary page

Commit:
Tidy up kube summary

nwmac committed:
Latest updates
